lib/elixir/lib/calendar/date.ex

```diff
@@ -57,6 +57,31 @@ defmodule Date do
                    day: Calendar.day, calendar: Calendar.calendar}
 
   @doc """
+  Returns a Date.Range
+
+  ## Examples
+
+      iex> Date.range(~D[2000-01-01], ~D[2001-01-01])
+      #DateRange<~D[2000-01-01], ~D[2001-01-01]>
+
+  """
+
+  @spec range(Date.t, Date.t) :: Date.Range.t
+  def range(%{calendar: first_calendar} = first, %{calendar: last_calendar} = last)
+    when first_calendar == last_calendar do
```

Comment:
I think 

`when first_calendar == last_calendar`

can be replaced by adjusting match: 

`def range(%{calendar: calendar} = first, %{calendar: calendar} = last)`